### PR TITLE
Multi tenant sample for v6

### LINKED ISSUE #1291: add v6 multitenant
http://docs.particular.net/samples/outbox/multi-tenant/

### PR DESCRIPTION
**Pull request by HEskandari**
I updated the multi-tenant sample for v6. Please note that the PipelineExecutor is not available anymore, so it looks a bit different. Also added a note about thread-safety issues on this sample.

fixes #1291

### CODE REVIEW

File: samples/outbox/multi-tenant/Version_6/Shared/ConfigErrorQueue.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+using NServiceBus.Config;
+using NServiceBus.Config.ConfigurationSource;
+
+internal class ConfigErrorQueue : IProvideConfiguration<MessageForwardingInCaseOfFaultConfig>
+{
+    public MessageForwardingInCaseOfFaultConfig GetConfiguration()
+    {
+        return new MessageForwardingInCaseOfFaultConfig
+        {
+            ErrorQueue = "error"
```
Comment (SimonCropp) [inline samples/outbox/multi-tenant/Version_6/Shared/ConfigErrorQueue.cs]: can u move over to the error code api

File: samples/outbox/multi-tenant/Version_6/Receiver/app.config [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    <add name="B" connectionString="Data Source=.\SQLEXPRESS;Initial Catalog=TenantB;Integrated Security=True"/>
+  </connectionStrings>
+  <UnicastBusConfig>
+    <MessageEndpointMappings>
+      <add Assembly="Shared" Endpoint="Samples.MultiTenant.Sender"/>
+    </MessageEndpointMappings>
+  </UnicastBusConfig>
+  <TransportConfig MaxRetries="0" />
+  <AuditConfig QueueName="audit"/>
+<startup><supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5.2"/></startup></configuration>
```
Comment (SimonCropp) [inline samples/outbox/multi-tenant/Version_6/Receiver/app.config]: remove `<supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5.2"/></startup>`

File: samples/outbox/multi-tenant/Version_6/Sender/app.config [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+<configuration>
+  <appSettings>
+    <add key="NServiceBus/Outbox" value="true" />
+    <add key="ClientSettingsProvider.ServiceUri" value="" />
+  </appSettings>
+  <connectionStrings>
+    <add name="NServiceBus/Persistence" connectionString="Data Source=.\SQLEXPRESS;Initial Catalog=Sender;Integrated Security=True" />
+  </connectionStrings>
+  <startup>
+    <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5.2" />
```
Comment (SimonCropp) [inline samples/outbox/multi-tenant/Version_6/Sender/app.config]: remove the startup node

File: samples/outbox/multi-tenant/Version_6/Shared/ConfigErrorQueue.cs (hunk 1 above)
Comment (HEskandari) [inline samples/outbox/multi-tenant/Version_6/Shared/ConfigErrorQueue.cs]: Sure, will do. Wanted to do a bit of 'house-keeping' but thought to leave it for now for people to review, as it is already different than v5.

**Comment by SimonCropp** [comment]
@HEskandari a couple of minor things to fix. but this looks good

I assume u smoke tested it? and i dont have to?

**Comment by HEskandari** [comment]
Yeah I did a smoke test, works fine now. There's an issue that I picked up that has to be applied to both v5 and v6. I'll fix that as well.

**Comment by SimonCropp** [comment]
cool. can u fix the minor comments and i will merge

**Comment by HEskandari** [comment]
@SimonCropp should be all good now.

File: samples/outbox/multi-tenant/Version_6/Receiver/OrderMap.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+using NHibernate.Mapping.ByCode;
+using NHibernate.Mapping.ByCode.Conformist;
+
+internal class OrderMap : ClassMapping<Order>
```
Comment (andreasohlund) [inline samples/outbox/multi-tenant/Version_6/Receiver/OrderMap.cs]: Why internal?

File: samples/outbox/multi-tenant/Version_6/Receiver/MultiTenantOpenSqlConnectionBehavior.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+        #endregion
+    }
+
+    public class Registration : RegisterStep
+    {
+        public Registration()
+            : base("OpenSqlConnection", typeof(MultiTenantOpenSqlConnectionBehavior), "Creates a new connnection for the tenant based on the Id on the header.")
+        {
+            InsertBefore("ExecuteUnitOfWork");
```
Comment (andreasohlund) [inline samples/outbox/multi-tenant/Version_6/Receiver/MultiTenantOpenSqlConnectionBehavior.cs]: Do we need this?

**Pull request merged**